src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -386,9 +389,10 @@ public void cleanup(CleanupStrategy strategy) {
           LOGGER.error(
               () -> String.format("Could not delete a local repository: %s", info.path()), e);
         }
-        localRepositoriesInfo.remove(url);
+        toBeRemoved.remove(url);
       }
     }
+    toBeRemoved.forEach(localRepositoriesInfo::remove);
```

Comment:
`Set.removeAll()` can also be used but not a big deal.

Reply:
Afaik `Set.removeAll()` can not be used at we want to remove the `URL`s from the map (which does not over a `removeAll()` method).